tests/components/gardena_bluetooth/test_button.py

```diff
@@ -0,0 +1,67 @@
+"""Test Gardena Bluetooth sensor."""
+
+
+from unittest.mock import Mock, call
+
+from gardena_bluetooth.const import Reset
+import pytest
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.button import DOMAIN as BUTTON_DOMAIN, SERVICE_PRESS
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+
+from . import setup_entry
+
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture
+def mock_switch_chars(mock_read_char_raw):
+    """Mock data on device."""
+    mock_read_char_raw[Reset.factory_reset.uuid] = b"\x00"
+    return mock_read_char_raw
+
+
+async def test_setup(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+    mock_entry: MockConfigEntry,
+    mock_switch_chars: dict[str, bytes],
+) -> None:
+    """Test setup creates expected entities."""
+
+    entity_id = "button.mock_title_factory_reset"
+    coordinator = await setup_entry(hass, mock_entry, [Platform.BUTTON])
```

Comment:
We shouldn't interact with integration details like the coordinator in the tests. Move time forward if needed to trigger a refresh.

Reply:
https://github.com/home-assistant/core/pull/96921